_posts/2019-03-11-PC-Engines_firmware-v4-9-0-3.md

```diff
@@ -0,0 +1,150 @@
+---
+layout: post
+title:  "PC Engines apu coreboot Open Source Firmware v4.9.0.3"
+date:   2019-03-11
+categories: Firmware
+---
+# PC Engines apu coreboot Open Source Firmware v4.9.0.3
+
+## Key changes
+
+Mainline:
+
+1. **Enabled** CPU **boost** feature in **runtime** configuration. Some
+   platforms seem to have problems with boost thus we decided to make CPU boost
+   runtime configurable.
+2. We have **upgraded** to coreboot **SDK 1.52** in mainline releases.
+3. Configured **pull-ups** on **WLAN_DISABLE#** pins on the mPCIe connectors
+   which could cause issues with certain modems when floating.
+4. **Reproducible builds limiation** has been **eliminated** by setting iPXE
+   and Memtest86+ fixed revisions. More details [here](https://github.com/pcengines/coreboot/issues/267)
+5. Uploaded SHA256 and signatures of all previous firmware releases. SHA256
+   files and signatures have been uploaded to pcengines.github.io.
+6. **Fixed microcode update** feature, which broke due to the upstream changes
+   in coreboot. More details [here](https://github.com/pcengines/apu2-documentation/issues/75#issuecomment-463067220)
+7. **Added** information about **ECC** memory capability in **SMBIOS tables**
+   on 4GB platforms.
+8. **Added interrupt** configuration entries for **PCIe bridge** devices 2.4
+   and 2.5.
```

Comment:
@miczyg1 does it have any business level value?

Reply:
@pietrushnic in some cases OS may use these settings for interrupt configuration. Please find corresponding JIRA item PE-468 for more details.

Reply:
@miczyg1 please improve description to be more clear about what it should fix - at this point it is very hard to understand why we doing this change.

Reply:
@pietrushnic its not about the fix, but to avoid a bug or wrong device operation. We could not find any negative impacts when the change was not included. But it is always better to have the interrupts programmed than to rely on OS that will possibly handle that.

Reply:
@miczyg1 IIUC we looking for `Error Correction Type: Multi-bit ECC`? It should be done explicit in video e.g. by grepping this sentence. I'm not sure you can do anything with that at this point. BTW I'm not sure whole booting procedure recording is nedded.

Reply:
Description improved